### PR TITLE
CLDR-12020 improve ExampleGenerator performance

### PR DESCRIPTION
**Pull request by btangmu**
-Extend ExampleGenerator cache to include both NATIVE and ENGLISH

-Remove ExampleContext class

-Remove ExampleBuilder.java

-Replace DataSection.examplebuilder with DataSection.nativeExampleGenerator

-DataSection call getExampleHtml for ExampleType.ENGLISH directly, no examplebuilder

-Remove 2-arg version of ExampleGenerator.getExampleHtml, use explicit ExampleType

-Fix testExampleGenerator null value bug

-Rename type to exType for clarity when it is ExampleType

-Disallow null ExampleType

-Change setUserAndFileForVotelist to setUserForVotelist without CLDRFile

-Remove some dead code

-Fix some warnings; formatting; comments

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-12020
- [x] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

**Comment by btangmu** [comment]
The plan is to postpone merging this until vetting phase is over and xml has been generated for this release. Also to review the changes with thread-safety in mind.

File: tools/java/org/unicode/cldr/test/ExampleGenerator.java [hunk 1]
```diff
@@ -260,22 +260,6 @@ public ExampleGenerator(CLDRFile resolvedCldrFile, CLDRFile englishFile, String
         NATIVE, ENGLISH
     };
 
-    public static class ExampleContext {
```
[the remark this answers is not in the document]
Comment (btangmu) [reply tools/java/org/unicode/cldr/test/ExampleGenerator.java]: There's a comment in the issue (dated July 16, 2019, 11:56 AM) where I wrote "ExampleContext serves no purpose..." and I included evidence there that it's unused. I'm not sure what purpose ExampleContext formerly served. In another comment I wrote "The function ExampleBuilder.getHelpHtml is never called. I’ll delete it." In general, I very frequently find cruft like this, presumably left over from features that no longer exist or implementations that have been replaced. Such cruft gets in the way of optimizations. When I realize that I'm trying to figure out how to cache an unused object, I'd rather clean up the code instead. After a bunch of clean-up, it became clear that ExampleBuilder was also essentially cruft and was best removed. These changes made it possible to improve some existing caching. There's still more to be done in terms of caching ExampleGenerator. I'm eager to get these changes in and then continue with that work.

File: tools/java/org/unicode/cldr/test/ConsoleCheckCLDR.java [hunk 2]
```diff
@@ -663,11 +660,8 @@ public static void main(String[] args) throws IOException {
                 String example = "";
 
                 if (SHOW_EXAMPLES) {
-                    example = ExampleGenerator.simplify(exampleGenerator.getExampleHtml(path, value, exampleContext,
-                        ExampleType.NATIVE));
-                    showExamples(checkCldr, prettyPath, localeID, path, value, fullPath, example,
-                        exampleContext);
-                    // continue; // don't show problems
+                    example = ExampleGenerator.simplify(exampleGenerator.getExampleHtml(path, value, ExampleType.NATIVE));
```
[the remark this answers is not in the document]
Comment (btangmu) [reply tools/java/org/unicode/cldr/test/ConsoleCheckCLDR.java]: OK, I'll try to save formatting changes for separate PRs.

**Comment by macchiati** [comment]
Doing them first to clean up the code is fine too...

Mark

File: tools/java/org/unicode/cldr/test/ExampleGenerator.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (macchiati) [reply tools/java/org/unicode/cldr/test/ExampleGenerator.java]: Ok, sounds good.

**Comment by btangmu** [comment]
I just fixed a merge conflict in submit.jsp. Unfortunately that undid Mark's review.

**Comment by srl295** [comment]
@btangmu shouldn't undo the review, at worst hides it (but i don't see it hidden).

**Pull request merged**